meson.build

```diff
@@ -222,6 +212,16 @@ config_h.set('HAVE_LIBSYSTEMD', enable_logind)
 
 config_h.set('HAVE_PIDFD_OPEN', cc.get_define('SYS_pidfd_open', prefix: '#include <sys/syscall.h>') != '')
 
+# systemd unit / sysuser file installation directories
+systemd_dep = dependency('systemd', not_found_message: 'systemd.pc pkg-config file required but not found')
+systemd_systemdsystemunitdir = get_option('systemdsystemunitdir')
+if systemd_systemdsystemunitdir == ''
+  # FIXME: systemd.pc file does not use variables with relative paths, so `define_variable` cannot be used
+  systemd_systemdsystemunitdir = systemd_dep.get_pkgconfig_variable('systemdsystemunitdir')
+endif
+
+systemd_sysusers_dir = systemd_dep.get_pkgconfig_variable('sysusers_dir', default: '/usr/lib/sysusers.d')
```

Comment:
This should at least provide an option to manually specify the directory like for system units, in case there is no libsystemd.

Reply:
This is unrelated to libsystemd, it's a separate and independent pkg-config file

Reply:
Ah, then in case there's no systemd.pc.

Reply:
Then it's a bug in the package/build script/etc and it needs to be provided

Reply:
It seems unnecessarily unkind to packagers to add a new required dependency just to get a single path.

Reply:
It's two, and one day it will be three, and then four, and then... - the point of pkg-config is exactly to avoid having to proliferate hundreds of build options in every single build system for every single project, and instead have one interface that can be used to customize things.
And it's for the benefits of the antis too, btw: it's just a text file with a common format, so they can just define it with `systemdsystemunitdir=/dev/null` or something like that et voila, these config files disappear automagically. win/win

Reply:
What about operating systems that systemd doesn't even run on but polkit does, like FreeBSD?  They have to create a systemd.pc file to configure the installation of a bunch of files that they're never going to use, rather than just having a way to not install these files that are irrelevant to them in the first place?

Reply:
Yes - again, that's something done _once_ that will work everywhere, for every project being built, instead of having to go around and ask to merge patches to add individual build options to each and every project, and then use them. How is this not better in every way? And if the answer is "because it has 'systemd' in the file name" then I'm going to yawn so hard I'll dislocate my jaw

Reply:
It's nothing to do with it being systemd — I like systemd, I've contributed to systemd, and I'd feel the same way for every other optional dependency.  I think it's uncharitable to assume that disagreement about the best way to handle optional features is some sort of bad faith "I don't like systemd" trolling.

I don't think it's better because it creates work where there could be none (just disabling the feature or using a default path if systemd.pc was not found, for example).  You've even already supplied a default value, but just written the code in a way so that it fails if the .pc file is present before it gets to the point where it could fall back to the default you've already supplied.  How is that better?

Reply:
It's not true that it creates more work, it removes it. The alternative is doing more work to add one build option for each path, which is going to be different for every project, find it, and use it. Versus just defining the stub .pc once and using it everywhere.

Reply:
Again the important bit is that this is not just about polkit, pretty much everything is starting to use sysusers and tmpfiles.d. Patching N projects to add 2 build options each does not scale well. Defining and installing a stub .pc is done once and works everywhere.

Reply:
But why is it acceptable to fall back to the default value if the .pc file is present but doesn't contain the required option, but it's not acceptable to fall back to the default value if the .pc file is not present?

Reply:
I don't mean to ad fuel to the fire but, TBH, configuring <dependency> if <dependency> is not (and probably never will be) present on the system (be it systemd or whatever) seems like abomination to me (/me just shot himself in foot, probably), Even though we use prepared and good-looking system like pkgconf. @bluca, this is nothing personal, I just believe there are cleaner ways than dropping unused config files in possibly-yet-non-existent directories on systems that have right to go different ways.

Reply:
Please don't force a stub .pc file. The existence of `systemd.pc` (even a stub) can imply that systemd exists on the system, and there will be programs that will make that assumption and configure themselves incorrectly.

Reply:
> I don't mean to ad fuel to the fire but, TBH, configuring 'dependency' if 'dependency' is not (and probably never will be) present on the system (be it systemd or whatever) seems like abomination to me (/me just shot himself in foot, probably), even though we use prepared and good-looking system like pkgconf. @bluca, this is nothing personal, I just believe there are cleaner ways than dropping unused config files in possibly-yet-non-existent directories on systems that have right to go different ways. But maybe I'm still a purist in this topic :D

I disagree, but it's your project, so as you wish - the pkgconf is now optional, and the well-known default installation directories are used as a fallback if missing.